### DISCUSSION TITLE
Provide package version

### DISCUSSION THREAD

**Discussion by bicarlsen** [PyO3/maturin#1415] [Q&A] [answered]
Is there a convenient way to provide the version of my package from the `pyproject.toml` file? e.g. `my_package.__version__`.

**Comment by konstin** [comment, marked as answer]
Is this something that could be solved by [importlib.metadata.version](https://docs.python.org/3/library/importlib.metadata.html#distribution-versions)? E.g. in `__init__.py`:

```python
from importlib.metadata import version

__version__ = version(__name__)
```

**Reply by bicarlsen** [reply]
Ah, sorry, I should have clarified.

I am trying to do this from the Rust code.

**Reply by konstin** [reply]
Are the version from Cargo.toml and pyproject.toml in sync? In that case https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates

**Reply by bicarlsen** [reply]
Exactly what I was looking for! Thank you :)

**Comment by bicarlsen** [comment]
Can I also ask, how do you assign a property to the package itself? e.g. `my_package.my_property`?

**Reply by messense** [reply]
Something like this: https://github.com/pydantic/pydantic-core/blob/25120edac6f9a17ce19125c53125c0c9d5411b52/src/lib.rs#L44

**Reply by bicarlsen** [reply]
Perfect. Thank you :)
